inspirehep/modules/migrator/cli.py

```diff
@@ -126,39 +126,54 @@ def remove_legacy_tables():
 
 
 @migrator.command()
+@with_appcontext
 def clean_records():
     """Truncate all the records from various tables."""
     from sqlalchemy.engine import reflection
+    from invenio_search import current_search, current_search_client
 
-    print('>>> Truncating all records.')
+    click.secho('>>> Truncating all records.')
 
     tables_to_truncate = [
-        "bibrec",
-        "record_json",
+        "records_metadata",
+        "records_metadata_version",
+        "inspire_prod_records",
+        "inspire_orcid_records",
+        "pidstore_pid",
     ]
     db.session.begin(subtransactions=True)
     try:
-        db.engine.execute("SET FOREIGN_KEY_CHECKS=0;")
-
-        # Grab any table with foreign keys to bibrec for truncating
+        # Grab any table with foreign keys to records_metadata for truncating
         inspector = reflection.Inspector.from_engine(db.engine)
         for table_name in inspector.get_table_names():
             for foreign_key in inspector.get_foreign_keys(table_name):
-                if foreign_key["referred_table"] == "bibrec":
+                if foreign_key["referred_table"] == "records_metadata":
                     tables_to_truncate.append(table_name)
 
-        if not prompt_bool("Going to truncate: {0}".format(
+        if not click.confirm("Going to truncate:\n{0}".format(
                 "\n".join(tables_to_truncate))):
             return
-        for table in tables_to_truncate:
-            db.engine.execute("TRUNCATE TABLE {0}".format(table))
-            print(">>> Truncated {0}".format(table))
 
-        db.engine.execute("DELETE FROM pidSTORE WHERE pid_type='recid'")
-        print(">>> Truncated pidSTORE WHERE pid_type='recid'")
+        click.secho('Truncating tables...', fg='red', bold=True,
+                    file=sys.stderr)
+        with click.progressbar(tables_to_truncate) as tables:
+            for table in tables:
+                db.engine.execute("TRUNCATE TABLE {0} CASCADE".format(table))
```

Comment:
You should add `RESTART IDENTITY` as per https://www.postgresql.org/docs/current/static/sql-truncate.html

Otherwise the sequences are not going to be reinitialized.